lib/internal/process/report.js

```diff
@@ -17,16 +18,29 @@ const report = {
     } else if (err === null || typeof err !== 'object') {
       throw new ERR_INVALID_ARG_TYPE('err', 'Object', err);
     }
+    const errorProperties = {};
+    const names = Object.getOwnPropertyNames(err);
+    for (const name of names) {
+      if (!['stack', 'message'].includes(name))
+        errorProperties[name] = `${err[name]}`;
+    }
 
-    return nr.writeReport('JavaScript API', 'API', file, err.stack);
+    return nr.writeReport('JavaScript API', 'API', file, err.stack,
+                          errorProperties);
```

Comment:
Instead of passing through an additional property, would it not be better to just pass through `err` directly? That way it would be possible to access the properties directly in the C++ layer and we'd need less arguments.

Reply:
I refactor the `error.stack` to `error` now